### PR TITLE
add migration to fix notifications from pre-1.0.0 upgrade

### PR DESCRIPTION
**Pull request by bmc-msft**
## Summary of the Pull Request

In pre-1.0.0, notifications mapped notification_id to RowKey and Container to PartitionKey.  In the 1.0.0 release, these got swapped causing pre-1.0.0 upgraded systems to fail.

## PR Checklist
* [ ] Applies to work item: #xxx
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

A migration for pre-1.0.0 keys in the notifications table

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

**Review by chkeita — APPROVED** [review]

File: src/deployment/data_migration.py [hunk 1]
```diff
@@ -48,3 +82,26 @@ def migrate(table_service: TableService, migration_names: List[str]) -> None:
         print("applying migration '%s'" % name)
         migrations[name](table_service)
         print("migration '%s' applied" % name)
+
+
+def main():
```
Comment (ranweiler) [inline src/deployment/data_migration.py:87]: It'd be nice to have some user-facing INFO-level logging so it is clear what's going on.

File: src/deployment/data_migration.py [hunk 2]
```diff
@@ -38,8 +42,38 @@ def migrate_task_os(table_service: TableService) -> None:
...
+def migrate_notification_keys(table_service: TableService) -> None:
+    table_name = "Notification"
+    notifications = table_service.query_entities(
+        table_name, select="PartitionKey,RowKey,config"
+    )
+    partitionKey = None
+
+    count = 0
+    for entry in notifications:
+        try:
```
Comment (ranweiler) [inline src/deployment/data_migration.py:54]: Could you add a comment about why this a non-exceptional UUID parse means we don't have to do anything?

File: src/deployment/data_migration.py [hunk 3]
```diff
@@ -38,8 +42,38 @@ def migrate_task_os(table_service: TableService) -> None:
     print("migrated %s rows" % count)
 
 
+def migrate_notification_keys(table_service: TableService) -> None:
```
Comment (ranweiler) [inline src/deployment/data_migration.py:45]: We should maybe have much more explicit names and/or comments, including version info, for migration functions.

**Review by ranweiler — APPROVED** [review]

File: src/deployment/data_migration.py (hunk 3 above)
Comment (bmc-msft) [reply src/deployment/data_migration.py:45]: Something to think about for the future.

**Pull request merged**